### DISCUSSION TITLE
Which is better, Matlab or Python?

### DISCUSSION THREAD

**Discussion by Fion-Lin** [1101-GS4719-C/Python-Community-GS4719-C#1586] [General]
我在系上的實驗室都在做一些訊號處理的project，大家都熟知的Matlab跟Python都是作圖及跑code的好媒介，但我一直在思考的問題是: 到底哪個軟體比較適合? 我自己之前是只有淺淺的碰過Python(大概是大一微積分的上機程度而已)，後來都是使用Matlab來完成project，因為Matlab還蠻直觀的，而且也有非常多套件可以使用。所以看完這篇文章後，我覺得他給了蠻好的解釋，把兩個語言好的地方都提到了，所以我覺得之後會想要試試看以Python作為工具來試試看!!(畢竟感覺之後在業界還是會比較需要Python，我要來好好認真學了)

[https://towardsdatascience.com/theres-a-clear-winner-between-matlab-vs-python-f6bb56b2b930](url)

**Comment by 1chooo** [comment]
我覺得 matlab 跟其他語言最大的不同就是：矩陣是從 1 開始數，不像其他程式語言是從 0 開始數，這點其實用起來很不直觀，常常看課本的邏輯做都會出錯，不過還是得說 matlab 畫出來的圖非常多樣，可以包含非常多的細節！

總之程式語言就是個工具，能做的出來都是好工具，沒有誰好誰壞！

**Comment by Yanxxx7979** [comment]
依我個人科系的淺薄使用來說，我覺得兩個系統的使用方向還滿不同的。matlab通常我們來用來跑系統模擬等等，而python 則會用來做相對更多的功能，包括計算等等

**Comment by davidteng00000** [comment]
我沒有使用過matlab，但是我覺得python有一個優點: 可以用big_integer大數運算，這是C/C++都做不到的事情(如果要在c++達成，只能換成string 慢慢處理，超麻煩)，不知道matlab有沒有大數運算
